packages/core/src/fhir-deduplication/resources/medication-request.ts

```diff
@@ -24,29 +45,41 @@ export function deduplicateMedRequests(medications: MedicationRequest[]): {
  */
 export function groupSameMedRequests(medRequests: MedicationRequest[]): {
   medRequestsMap: Map<string, MedicationRequest>;
-  remainingMedRequests: MedicationRequest[];
   refReplacementMap: Map<string, string[]>;
 } {
   const medRequestsMap = new Map<string, MedicationRequest>();
   const refReplacementMap = new Map<string, string[]>();
-  const remainingMedRequests: MedicationRequest[] = [];
+
+  function assignMostDescriptiveStatus(
+    master: MedicationRequest,
+    existing: MedicationRequest,
+    target: MedicationRequest
+  ): MedicationRequest {
+    master.status = pickMostDescriptiveStatus(statusRanking, existing.status, target.status);
+    return master;
+  }
 
   for (const medRequest of medRequests) {
     const medRef = medRequest.medicationReference?.reference;
-    const status = medRequest.status;
+    const date = medRequest.authoredOn;
```

Comment:
https://github.com/metriport/metriport/pull/2633
this PR should probably be upstream of that right?